ci/test.sh

```diff
@@ -118,3 +118,77 @@ assert_no_diff
 
 # Run the integration tests
 (cd bindgen-integration && cargo test $CARGO_ARGS)
+
+if [ "$BINDGEN_RUST_FOR_LINUX_TEST" == "1" ]; then
+  # Run the Rust for Linux test
+  #
+  # This is intended to be an end-to-end test that runs what Linux kernel
+  # developers/users would do. It is a quick, build-only test of the Rust code
+  # in the Linux kernel.
+
+  # Put LLVM binaries in the path for `LLVM=1`. The LLVM `bin` directory should
+  # go first since there are others in the Ubuntu image.
+  export PATH="${LLVM_DIRECTORY}/bin:${PATH}"
+
+  # Kernel build dependency: `bindgen-cli`, which is under test.
+  #
+  # Using the unstable `--out-dir` so that we can use `cargo build` instead of `cargo install`,
+  # since we want to reuse the `$CARGO_ARGS`.
+  (cd bindgen-cli && RUSTC_BOOTSTRAP=1 cargo build -Zunstable-options --out-dir ${HOME}/.bindgen $CARGO_ARGS)
```

Comment:
It seems you wouldn't need `RUSTC_BOOTSTRAP` etc if you use `CARGO_TARGET_DIR=$HOME/.bindgen` or so?

https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-reads

Reply:
Apparently the `--target-dir` flag is also not unstable.

Reply:
Ah, ok so I see what you're doing here, you just want to make sure you get the `target-dir/debug/` or release right...

Maybe given we're only interested on running this on linux (so no explicit `--target`), and we know whether we're doing a release build (via `BINDGEN_RELEASE_BUILD`), maybe just `cargo build $CARGO_ARGS`, and then either conditionally add `target/debug` or `target/release` to the `PATH`, or maybe just add both to the PATH?

Reply:
Yeah, I did this to be able to use whatever you had in `$CARGO_ARGS` (and `cargo install` does not accept e.g. `--release`), and it seemed like the clean way, but sadly unstable.

The `--target` should not matter for this CI as you say, but I guess you could be adding different profiles in the future. If that is not a concern, then simply adding both `debug/release` to `$PATH` sounds the simplest.

(It could still "break" or be surprising if other things get refactored and both debug and release get built in the same test for something else above. We could use `--target-dir` to at least isolate that).